packages/frontend/src/contexts/user.js

```diff
@@ -0,0 +1,42 @@
+import { makeAutoObservable } from 'mobx'
+import { SERVER } from '../config'
+
+export default class User {
+  commitment =
+    '21148151481457093107206483541042547669092147310094944251743153632587065177648'
+  epoch = ''
+  attesterId = ''
+  id = ''
+  signUps = []
+
+  constructor(state) {
+    makeAutoObservable(this)
+    this.state = state
+    if (typeof window !== 'undefined') {
+      this.load()
+    }
+  }
+
+  async load() {
+    // have not yet found how to pass :userId when page /user/:id is rendered
+    const url = new URL(`api/user/${this.commitment}`, SERVER)
+    const response = await fetch(url.toString())
+    // this works to get a single sign-up with findOne:
+    const { epoch, attesterId, _id } = await response.json()
+    this.epoch = epoch
+    this.attesterId = attesterId
+    this.id = _id
+
+    // below gets a list of sign-ups with findMany,
+    // can log these values from here but trying to get them with:
+    // user.signUps[0] from UserPage returns undefined.
+
+    // this.signUps = await response.json()
+    // console.log(this.signUps)
+    // console.log(this.signUps[0])
+    // console.log(this.signUps[0].attesterId)
+    // console.log(this.signUps[0].epoch)
+    // console.log(this.signUps[0]._id)
+    // console.log(this.signUps.length)
+  }
+}
```

Comment:
This issue is hanging me up. Atm my routes are only returning single objects (single events from the db). I am able to use `findMany()` to return an array of objects (can access the array, its index, and the values of each object from here), but these same logs on `UserPage.tsx`
- `console.log(this.signUps)` -->  can see same array as here (also can get `signUps.length`)
-  `console.log(this.signUps[0])` -->  can see same object as here, np yet but then..
* * *
- `console.log(this.signUps)` -->  can see same array as here (also can still get `signUps.length`)
-  `console.log(this.signUps[0])` -->  undefined
-  `console.log(this.signUps[0].attesterId)` --> error, as [0] is now undefined

So I am able to get arrays of events but not display them. Any ideas?

Reply:
Ok I'm going to leave a sorta general comment here. The routes in the backend look good, but on the frontend your approach to contexts is not correct. You don't necessarily want one context per route, you want to group contexts by logic. So if you have routes for getting attestations maybe you have a context that handles:
- `/api/attestations`
- `/api/attestations/:id`
- `/api/epoch/:number/attestation`

Something like that. How much logic you put in a single context is up to you.

Each of these contexts are **singletons**. Only one instance of the class will exist for the lifetime of the application. The `load` function is called when the instance is created, so that it can initialize itself however it needs to. You _can_ load data here if you _know_ the user will want to see it immediately, otherwise better to load when they need it. The `load` function is called once and never again.

As for handling the data, [here](https://github.com/vimwitch/canon/blob/main/packages/frontend/src/contexts/canon.js) is an example building a store that loads a list of text sections that are displayed to the user. Each section of text has some metadata that is also loaded (the canon). You want to expose some simple functions so that your components can do things like `attestationContext.load()`, or `attestationContext.loadPage(x)`. You generally want to write these functions as the application needs them.

Then you can see [here](https://github.com/vimwitch/canon/blob/main/packages/frontend/src/Home.jsx#L55) how we use that section data. To answer a question you asked somewhere, yes, you can simply iterate over the data you have if you're showing a list. Note that in that example I'm passing the `sectionId`, not the full section object.

I can load the full section in the component [here](https://github.com/vimwitch/canon/blob/main/packages/frontend/src/components/CanonSection.jsx#L14) by accessing that `sectionsById` property on the store. Note that the server returns an array, then the frontend builds a mapping of `id => object` so the app can easily access objects. You'll want to build these for each type of data you're working with.

Because we're using mobx if I load the sections again and a section has changed, the context will update `sectionsById`, which will cause any components referencing a section from this object to update automatically. So we can pull new data and update the ui without having to write code to explicitly do it.

Reply:
Okay great, mostly I understand; will digest and review. 
Thank you for taking the time, these explanations and examples are super valuable 🙏🏻